### PR TITLE
python: support ChannelError from AsyncChannel.run()

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
We add a function called run_wrapper() to catch ChannelError from the run() function and turn it into a close() message.

It would be more elegant to handle this by setting a done callback on the task and calling .result() from it in order to re-raise and catch the exception.  Unfortunately, there are some non-ChannelError exceptions being raised in our code right now, and they're currently being ignored, and this change would result in them being re-raised.  We can address that in a future PR.

### CODE REVIEW

File: src/cockpit/channel.py [hunk 1]
```diff
@@ -194,7 +201,7 @@ def do_pong(self, message):
     def do_open(self, options):
         self.receive_queue = asyncio.Queue()
         self.flow_control = options.get('flow-control') is True
-        asyncio.create_task(self.run(options), name='f{self.__class__.__name__}.run({options})')
+        asyncio.create_task(self.run(options), name='f{self.__class__.__name__}.run_wrapper({options})')
```
Comment (mvollmer) [inline src/cockpit/channel.py:204]: Also `self.run_wrapper(options)` in the first argument?

**Review by mvollmer — CHANGES_REQUESTED** [review]

File: src/cockpit/channel.py (hunk 1 above)
Comment (allisonkarlitskaya) [reply src/cockpit/channel.py:204]: oops :)

Comment (mvollmer) [reply src/cockpit/channel.py:204]: And maybe just `run` in the name?

**Review by mvollmer — APPROVED** [review]
Works great now!

**Pull request merged**